internal/controller/volsync/vshandler.go

```diff
@@ -2022,6 +2031,13 @@ func (v *VSHandler) createReadOnlyPVCFromSnapshot(rd *volsyncv1alpha1.Replicatio
 		}
 	}
 
+	volumeMode := corev1.PersistentVolumeFilesystem
+	if util.IsRBDEnabledForVolSyncReplication(v.owner.GetAnnotations()) &&
+		rdSpec.ProtectedPVC.VolumeMode != nil &&
+		*rdSpec.ProtectedPVC.VolumeMode == corev1.PersistentVolumeBlock {
+		volumeMode = corev1.PersistentVolumeBlock
+	}
```

Comment:
This repeats the same code twice instead of using a helper method.

Reply:
done